search_log.go

```diff
@@ -136,34 +177,45 @@ func readLine(reader *bufio.Reader) (string, error) {
 	return string(line), nil
 }
 
-func readLastLine(file *os.File) string {
-	var line []byte
-	var cursor int64
-	stat, _ := file.Stat()
-	filesize := stat.Size()
+// Read lines from the end of a file
+// endCursor initial value should be the filesize
+func readLastLines(file *os.File, endCursor int64) string {
+	var lines []byte
+	var cursor = endCursor
 	for {
-		cursor -= 1
-		file.Seek(cursor, io.SeekEnd)
-
-		char := make([]byte, 1)
-		file.Read(char)
-
-		// stop if we find a line
-		if cursor != -1 && (char[0] == 10 || char[0] == 13) {
+		// stop if we are at the begining
+		// check it in the start to avoid read beyond the size
+		if cursor <= 0 {
 			break
 		}
-		line = append(line, char[0])
-		if cursor == -filesize { // stop if we are at the begining
-			break
+
+		var size int64 = 512
```

Comment:
Read 512 bytes once instead of 1 byte

Reply:
I like this change very much, would you like to write a benchmark for this optimization?

Reply:
Comparison:

```
Old way:
searchLogSuite.BenchmarkReadLastLine        10000            124423 ns/op
searchLogSuite.BenchmarkReadLastLine        10000            126135 ns/op

New way:
searchLogSuite.BenchmarkReadLastLines      1000000              1892 ns/op
searchLogSuite.BenchmarkReadLastLines      1000000              1920 ns/op
```

Conclusion: the new implementation is 65x (126135/1920) as fast as the old way.

The benchmark for the readLastLine in PR #20